api/groups_test.go

```diff
@@ -2351,5 +2352,324 @@ func TestAddUserToGroup(t *testing.T) {
+					l, _ := strconv.Atoi(string(*input.GroupName)[len(*input.GroupName)-1:])
+					return ListGroupsUsers(l), nil
+				},
+				func(successResponse *models.SuccessResponse, errorResponse *models.ErrorResponse) {
+					var testOutJSON []models.ListGroupUsersType
+					jsonErr := json.Unmarshal(successResponse.Body, &testOutJSON)
+					So(jsonErr, ShouldBeNil)
+					So(testOutJSON, ShouldBeEmpty)
+
+					testOutCSV := string(successResponse.Body[:])
+					stringSlice := strings.Split(testOutCSV, "\n")
+					So(stringSlice, ShouldHaveLength, 1)
+					So(stringSlice[0], ShouldNotBeEmpty)
+
+					So(successResponse.Status, ShouldEqual, http.StatusOK)
+					So(errorResponse, ShouldBeNil)
+				},
+			},
+			{
+				"200 response - 1 group",
+				func(input *cognitoidentityprovider.ListGroupsInput) (*cognitoidentityprovider.ListGroupsOutput, error) {
+					output := listGroups(1)
+					return &output, nil
+				},
+				func(input *cognitoidentityprovider.ListUsersInGroupInput) (*cognitoidentityprovider.ListUsersInGroupOutput, error) {
+					l, _ := strconv.Atoi(string(*input.GroupName)[len(*input.GroupName)-1:])
+					return ListGroupsUsers(l + 1), nil
+				},
+				func(successResponse *models.SuccessResponse, errorResponse *models.ErrorResponse) {
+					var testOutJSON []models.ListGroupUsersType
+					jsonErr := json.Unmarshal(successResponse.Body, &testOutJSON)
+					So(jsonErr, ShouldBeNil)
+					So(testOutJSON, ShouldNotBeEmpty)
+					So(testOutJSON, ShouldHaveLength, 1)
+					for _, t := range testOutJSON {
+						So(t.GroupName, ShouldResemble, "group 0 description")
+						So(t.UserEmail, ShouldResemble, "user0.email@domain.test")
+					}
+					testOutCSV := string(successResponse.Body[:])
+					stringSlice := strings.Split(testOutCSV, "\n")
+					So(stringSlice, ShouldHaveLength, 1)
+					So(stringSlice[0], ShouldNotBeEmpty)
+					So(successResponse.Status, ShouldEqual, http.StatusOK)
+					So(errorResponse, ShouldBeNil)
+				},
+			},
+			{
+				"200 response - 3 groups",
+				func(input *cognitoidentityprovider.ListGroupsInput) (*cognitoidentityprovider.ListGroupsOutput, error) {
+					output := listGroups(3)
+					return &output, nil
+				},
+				func(input *cognitoidentityprovider.ListUsersInGroupInput) (*cognitoidentityprovider.ListUsersInGroupOutput, error) {
+					l, _ := strconv.Atoi(string(*input.GroupName)[len(*input.GroupName)-1:])
+					return ListGroupsUsers(l + 1), nil
+				},
+				func(successResponse *models.SuccessResponse, errorResponse *models.ErrorResponse) {
+					var testOutJSON []models.ListGroupUsersType
+					jsonErr := json.Unmarshal(successResponse.Body, &testOutJSON)
+					So(jsonErr, ShouldBeNil)
+					So(testOutJSON, ShouldNotBeEmpty)
+					So(testOutJSON, ShouldHaveLength, 6)
+					testOutCSV := string(successResponse.Body[:])
+					stringSlice := strings.Split(testOutCSV, "\n")
+					So(stringSlice, ShouldHaveLength, 1)
+					So(stringSlice[0], ShouldNotBeEmpty)
```

Comment:
I'm not sure we need the additional tests to check it's not in csv format if we've successfully unmarshalled the json.